### PR TITLE
Ports: Add GNU cpio archive utility port

### PR DESCRIPTION
**Pull request by supercomputer7**

### CODE REVIEW

File: Ports/cpio/patches/0001-Use-global-program_name-variable-from-gnu-dir.patch [hunk 1]
```diff
@@ -0,0 +1,31 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Liav A <liavalb@gmail.com>
+Date: Sat, 25 Jun 2022 05:45:29 +0300
+Subject: [PATCH] Use global program_name variable from gnu dir
```
Comment (timschumi) [inline Ports/cpio/patches/0001-Use-global-program_name-variable-from-gnu-dir.patch:4]: Is there a particular reason for doing this though?

File: Ports/cpio/package.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=cpio
+version=2.13
+useconfigure="true"
+use_fresh_config_sub="true"
```
Comment (timschumi) [inline Ports/cpio/package.sh:5]:
```suggestion
port='cpio'
version='2.13'
useconfigure='true'
use_fresh_config_sub='true'
```

File: Ports/cpio/package.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+workdir="cpio-${version}"
```
Comment (timschumi) [inline Ports/cpio/package.sh:6]: This is the default value and can be removed.

File: Ports/cpio/package.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+config_sub_paths=("build-aux/config.sub")
+files="https://ftp.gnu.org/gnu/cpio/cpio-${version}.tar.gz cpio-${version}.tar.gz
+https://ftp.gnu.org/gnu/cpio/cpio-${version}.tar.gz.sig cpio-${version}.tar.gz.sig
+https://ftpmirror.gnu.org/gnu/gnu-keyring.gpg gnu-keyring.gpg"
+auth_type="sig"
```
Comment (timschumi) [inline Ports/cpio/package.sh:11]: Same as with `gzip`, authentication via `sha256` would likely be preferred.

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Ports/cpio/patches/0001-Use-global-program_name-variable-from-gnu-dir.patch (hunk 1 above)
Comment (supercomputer7) [reply Ports/cpio/patches/0001-Use-global-program_name-variable-from-gnu-dir.patch:4]: It won't compile without this.

Comment (supercomputer7) [reply Ports/cpio/patches/0001-Use-global-program_name-variable-from-gnu-dir.patch:4]: I added a brief description to ensure it is clear why we need this patch.

**Review by timschumi — APPROVED** [review]
LGTM as well!

**Pull request merged**